File: fix-missing-exports.js
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';

// Function to fix missing exports in a file
function fixMissingExports(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Check if file already has export default
    if (content.includes('export default')) {
      return false; // Already has export
    }
    
    // Check if file has a React component function
    const componentMatch = content.match(/const\s+(\w+Page):\s+React\.FC\s*=\s*\(\)\s*=>\s*{/);
    if (!componentMatch) {
      return false; // No component found
    }
    
    const componentName = componentMatch[1];
    console.log(`Adding missing export to: ${filePath} (${componentName})`);
    
    // Add export default at the end of the file
    const lines = content.split('\n');
    const lastLine = lines[lines.length - 1];
    
    if (lastLine.trim() === '') {
      // Remove empty last line
      lines.pop();
    }
    
    // Add export statement
    lines.push('');
    lines.push(`export default ${componentName};`);
    
    const fixedContent = lines.join('\n');
    fs.writeFileSync(filePath, fixedContent, 'utf8');
    
    return true; // File was fixed
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Function to find all TypeScript/JavaScript files
function findSourceFiles(dir) {
  const files = [];
  
  function traverse(currentDir) {
    const items = fs.readdirSync(currentDir);
    
    for (const item of items) {
      const fullPath = path.join(currentDir, item);
      const stat = fs.statSync(fullPath);
      
      if (stat.isDirectory()) {
        // Skip node_modules and other irrelevant directories
        if (!['node_modules', '.git', 'dist', 'build', '.next', 'out'].includes(item)) {
          traverse(fullPath);
        }
      } else if (stat.isFile()) {
        // Check for TypeScript/JavaScript files
        if (/\.(ts|tsx|js|jsx)$/.test(item)) {
          files.push(fullPath);
        }
      }
    }
  }
  
  traverse(dir);
  return files;
}

// Main execution
console.log('Starting missing export fix...');

const srcDir = path.join(process.cwd(), 'src');
const files = findSourceFiles(srcDir);

let fixedCount = 0;

for (const file of files) {
  if (fixMissingExports(file)) {
    fixedCount++;
  }
}

console.log(`\nMissing export fix complete!`);
console.log(`Files processed: ${files.length}`);
console.log(`Files with added exports: ${fixedCount}`);

File: fix_missing_exports.py
#!/usr/bin/env python3
"""
Script to fix missing default exports in component files
"""
import os
import glob

def fix_missing_export(file_path):
    """Fix missing default exports in component files"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file has no default export
        if 'export default' not in content and 'export {' not in content:
            print(f"Fixing missing export in: {file_path}")
            
            # Extract component name from file path
            component_name = os.path.basename(file_path).replace('.tsx', '').replace('.ts', '')
            
            # Create a basic component with default export
            basic_component = f"""import React from 'react';

interface {component_name}Props {{
  children?: React.ReactNode;
}}

const {component_name}: React.FC<{component_name}Props> = ({{ children }}) => {{
  return (
    <div className="{component_name.lower()}">
      {{children}}
    </div>
  );
}};

export default {component_name};"""
            
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(basic_component)
            
            return True
        
        return False
        
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    """Main function to fix all missing exports"""
    # Get all component files
    component_files = []
    
    for root, dirs, files in os.walk('./app/components'):
        for file in files:
            if file.endswith(('.tsx', '.ts')):
                file_path = os.path.join(root, file)
                component_files.append(file_path)
    
    print(f"Found {len(component_files)} component files")
    
    fixed_count = 0
    for file_path in component_files:
        if fix_missing_export(file_path):
            fixed_count += 1
            print(f"Fixed: {file_path}")
    
    print(f"Fixed {fixed_count} component files with missing exports")

if __name__ == "__main__":
    main()